src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax;
+
+public class Main {
+    public static void main(String[] args) {
+        Shape[] shapes = new Shape[6];
+
+        for (int i = 0; i < shapes.length / 2; i++) {
+            shapes[i] = new ShapeSupplier().getRandomShape();
+            shapes[i].draw();
+        }
+        for (int i = shapes.length / 2; i < shapes.length; i++) {
+            shapes[i] = new ShapeSupplier().getDefaultShape();
```

Comment:
Similar to the previous loop, avoid creating a new `ShapeSupplier` instance inside the loop. Create it once before the loop starts and reuse it.